### ISSUE TITLE
Browser back button randomly failing on v3

### ISSUE THREAD

**Issue by djanowski** [framework7io/framework7#2917]
* Framework7 version: 3.6.2
* Platform and Target: Chrome browser
* Live Link or JSFiddle: Unfortunately can't reproduce consistently

### Describe the bug

Some times clicking the browser's back button fails:

```
Uncaught TypeError: Cannot read property 'route' of undefined
    at Router.back (back.js?ba6d:513)
    at afterAnimation (back.js?ba6d:304)
    at eval (back.js?ba6d:326)
    at onDone (router-class.js?113f:217)
    at HTMLDivElement.eval (router-class.js?113f:221)
    at HTMLDivElement.fireCallBack (dom7.module.js?f099:467)
    at HTMLDivElement.handleEvent (dom7.module.js?f099:333)
```

This is the line that fails:

https://github.com/framework7io/framework7/blob/e5c7c9e352a1f63f86c97c414ef67a6706294cdc/src/core/modules/router/back.js#L509

We don't know why `$previousPage[0].f7Page` is undefined, but a few lines above, the code is checking for whether `$previousPage[0].f7Page` is set or not:

https://github.com/framework7io/framework7/blob/e5c7c9e352a1f63f86c97c414ef67a6706294cdc/src/core/modules/router/back.js#L499

Maybe you can help us in figuring out why `f7Page` is unset? If it's an expected condition (because it's checked for a few lines above), then maybe this piece of code is missing the check?

Thank you very much in advance.

**Comment by nolimits4web** [comment]
Would be good to see live example or JSFiddle where i can see the issue

**Comment by djanowski** [comment]
I completely understand. We're working on isolating this issue into a standalone app that we can share with you.

In the meantime, any thoughts on why `$previousPage[0].f7Page` is checked for undefined on line 499 but not on 509?

**Comment by nolimits4web** [comment]
Actually that check is not really needed. Issue is that if there is no `f7Page` then it was added to DOM/loaded in some wrong way. Because if it is correctly added to DOM with router then it must be there, so issue is somewhere else. There shouldn't be previous page in DOM without `f7Page` property

**Comment by krolow** [comment]
So we have isolated this issue as @djanowski mention, and by isolation we could identify that the reason was kind of what u @nolimits4web have answered, we do have some async process that was happening before the page render it took a few ms but it seems that f7 always need one `<div class="page">` to be there... we have moved the async process to happen inside the (react page) and so render the `class="page-content"` and it fixed our issues with  the router.

So basically we always need to keep one `<Page>` mounted/rendered.

I think we can close that @djanowski 👍

**Comment by nolimits4web** [comment]
Yes, Page element must be always presented, and all its internals can be loaded async
